File: security/src/main/java/com/alstn113/security/app/ui/TestController.java
package com.alstn113.security.app.ui;

import com.alstn113.security.security.annotation.AuthenticationPrincipal;
import com.alstn113.security.security.authentication.JwtAuthentication;
import com.alstn113.security.security.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class TestController {

    @GetMapping("/public")
    public String test(@AuthenticationPrincipal JwtAuthentication authentication) {
        if (authentication == null) {
            return "Public endpoint.";
        }
        return "Hello #" + authentication.principal();
    }

    @PostMapping("/public")
    public String testPost(@AuthenticationPrincipal JwtAuthentication authentication) {
        return "Hello #" + authentication.principal();
    }

    @GetMapping("/private/member")
    public String privateMember(@AuthenticationPrincipal JwtAuthentication authentication) {
        return "Member #" + authentication.principal();
    }

    @GetMapping("/private/member/holder")
    public String privateMemberHolder() {
        JwtAuthentication authentication = (JwtAuthentication) SecurityContextHolder.getContext().getAuthentication();
        return "Member #" + authentication.principal();
    }

    @GetMapping("/private/admin")
    public String privateAdmin(@AuthenticationPrincipal JwtAuthentication authentication) {
        return "Admin #" + authentication.principal();
    }
}

File: refresh-token-rotation/refresh-token-rotation/src/main/kotlin/io/github/alstn113/rtr/adapter/in/web/controller/TestController.kt
package io.github.alstn113.rtr.adapter.`in`.web.controller

import io.github.alstn113.rtr.adapter.`in`.web.common.ApiResponse
import io.github.alstn113.rtr.adapter.`in`.web.security.AccountPrincipal
import io.github.alstn113.rtr.adapter.`in`.web.security.resolver.AuthenticationPrincipal
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class TestController {

    @GetMapping("/test")
    fun test(
        @AuthenticationPrincipal principal: AccountPrincipal,
    ): ResponseEntity<ApiResponse<String>> {
        val message = if (principal.isAnonymous()) {
            "안녕하세요! 당신은 익명 사용자입니다."
        } else {
            "안녕하세요! 사용자(${principal.accountId})님 환영합니다."
        }

        return ResponseEntity.ok(ApiResponse.success(message))
    }
}